### ISSUE TITLE
Params containing slashes break the `from` prop

### ISSUE THREAD

**Issue by jakst** [TanStack/router#1762]
### Describe the bug

When the value of `$var` contains a slash, the `$var.tsx` route will break if it contains a `from="/$var"` prop.

```tsx
// index.tsx
<Link to="/$var" params={{ var: 'some/thing' }}>...</Link>

// $var.tsx
<Link from="/$var" to="/$var">...</Link>
```

The issue was introduced in `v1.35.6` of the router

### Your Example Website or App

https://stackblitz.com/edit/github-d3fgdr?file=src%2Froutes%2F%24var.tsx

### Steps to Reproduce the Bug or Issue

1. Go to Stackblitz
2. Click "Go to problem"
3. Observer crash with message
```
Invariant failed: Could not find match for from: /$var
```

### Expected behavior

I expect URLs with params containing encoded slashes to not break routes.

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by mirague** [comment]
Was this also fixed by #1799 ?

**Comment by schiller-manuel** [comment]
yes
